Kernel/InterruptHandler.cpp

```diff
@@ -0,0 +1,53 @@
+#include <Kernel/InterruptHandler.h>
+#include <Kernel/SharedInterruptHandler.h>
+
+InterruptHandler::InterruptHandler(u8 irq)
+    : m_irq_number(irq)
+{
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).register_interrupt_handler(*this);
+    disable_irq();
+}
+
+InterruptHandler::~InterruptHandler()
+{
+    disable_irq();
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).unregister_interrupt_handler(*this);
```

Comment:
There is so much repetition here. Let's find a more concise way to express the same thing.. how about:

```cpp
SharedInterruptHandler::from_irq(m_irq_number).unregister(*this);
```

Reply:
I changed it to - 
```SharedInterruptHandler::from(m_irq_number).register_handler(*this);```
What do you think about it?